crates/sui-core/src/storage.rs

```diff
@@ -480,6 +482,27 @@ impl RestStateReader for RestReadStore {
             .map_err(StorageError::custom)
     }
 
+    fn get_lowest_available_checkpoint(
+        &self,
+    ) -> sui_types::storage::error::Result<CheckpointSequenceNumber> {
+        let object_watermark = self
+            .state
+            .get_object_cache_reader()
+            .get_highest_pruned_checkpoint()
+            .map(|seq| seq + 1)
+            .map_err(StorageError::custom)?;
+
+        let transaction_watermark = self.get_lowest_available_checkpoint_transactions()?;
+
+        Ok(std::cmp::max(object_watermark, transaction_watermark))
```

Comment:
why is it max?
if both pruners are available then transaction_watermark <= object_watermark, so for that delta effects/events are not present

Reply:
is there ever a situation where transaction pruning is on but objects isn't?

The intent here is to be more conservative with which watermark to return.

Reply:
>so for that delta effects/events are not present

I think its the reverse of that. The `transaction_watermark` shows the lowest checkpoint that we have transactions, effects, events available for while the `object_watermark` shows the lowest checkpoint that we have objects for. So if you want the value returned here to be the lowest checkpoint at which you have full data availability across transactions, effects, events, checkpoints and objects, you'd need to take the higher of the two right?

Reply:
> is there ever a situation where transaction pruning is on but objects isn't?

we don't do that, but I've seen it in the past multiple such configs from different FN providers 

> transaction_watermark <= object_watermark

yeah, sorry, objects are missing for the delta, not transactions. 
So it's objects pruner watermark if defined/set, otherwise it's checkpoint watermark.
But I guess if either one of them is not set, max will still return a correct result.
What do we want to return if both are 0?

Reply:
reflecting on this conversation i realized that it probably makes more sense to invert and possibly decouple them so now `lowest_available_checkpoint` is the transactions watermark and `lowest_available_checkpoint_objects` is the obejct_watermark.

Reply:
yes, might be less confusing:)